sklearn/preprocessing/_function_transformer.py

```diff
@@ -1,20 +1,30 @@
 import warnings
+from numbers import Number
 
 import numpy as np
 
 from ..base import BaseEstimator, TransformerMixin, _fit_context
 from ..utils._param_validation import StrOptions
-from ..utils._set_output import _get_output_config
+from ..utils._set_output import ADAPTERS_MANAGER, _get_output_config
 from ..utils.metaestimators import available_if
 from ..utils.validation import (
     _allclose_dense_sparse,
     _check_feature_names_in,
+    _get_feature_names,
     _is_pandas_df,
     _is_polars_df,
     check_array,
 )
 
 
+def _get_adapter_from_container(container):
+    module_name = container.__module__.split(".")[0]
```

Comment:
hopefully we don't get objects with no `__module__`. `pickle.whichmodule` is a lot safer, but we can leave this as is.

Reply:
Actually, `__module__` is on the `__class__` rather than on the instance. It was failing for numpy ;)